h2/src/main/org/h2/mvstore/tx/TransactionMap.java

```diff
@@ -116,7 +128,7 @@ public long sizeAsLong() {
                     K key = cursor.next();
                     // cursor.getValue() returns outdated value
                     VersionedValue data = map.get(key);
-                    data = getValue(key, readLogId, data);
+                    data = getValue(mapRootPage, undoRootPage, key, readLogId, data, committingTransactions);
```

Comment:
I replaced `cursor.getValue()` with `map.get(key)` here too in #954 for symmetry with the similar fix in `entryIterator()` and placed a comment about `cursor.getValue()` as a reminder.

`TMIterator.fetchNext()` inroduced in your PR uses `cursor.getValue()` again. May be this method can now be safely used here too?